integration/tests/cook/test_basic.py

```diff
@@ -845,3 +845,33 @@ def get_docker_info():
             self.assertEqual(instance['ports'][1], int(ports['9090/udp'][0]['HostPort']))
         finally:
             util.session.delete('%s/rawscheduler?job=%s' % (self.cook_url, job_uuid))
+
+
+    def test_unscheduled_jobs(self):
+        job_uuid, resp = util.submit_job(self.cook_url,
+                                         command='ls',
+                                         constraints=[['HOSTNAME', 'EQUALS', 'fakehost']])
+        self.assertEqual(resp.status_code, 201, resp.content)
+        try:
+            unscheduled_resp = util.session.get('%s/unscheduled_jobs?job=%s' % (self.cook_url,
+                                                                                job_uuid))
+            self.assertEqual(200, unscheduled_resp.status_code, unscheduled_resp)
+            unscheduled_resp = unscheduled_resp.json()[0]
+            self.assertTrue(
+                any(['The job is now under investigation. Check back in a minute for more details!' == reason['reason'] for reason in unscheduled_resp['reasons']]),
```

Comment:
Would there ever be more than 1 reason here? Should we assert `len(reasons) == 1`?

Reply:
If multiple tests run at once, there can be a separate reason that "You have jobs ahead of this in the queue" but that is nondeterministic so I opted for the `any` approach

Reply:
Can we add a comment explaining this?

Reply:
I think this comment may have been hidden due to other commits :point_up:

Reply:
done